app/code/Magento/CatalogInventory/Model/StockStateProvider.php

```diff
@@ -119,13 +119,13 @@ public function checkQuoteItemQty(StockItemInterface $stockItem, $qty, $summaryQ
         $result->setItemIsQtyDecimal($stockItem->getIsQtyDecimal());
         if (!$stockItem->getIsQtyDecimal()) {
             $result->setHasQtyOptionUpdate(true);
-            $qty = (int) $qty;
+            $qty = (int) $qty ?: 1;
             /**
              * Adding stock data to quote item
              */
             $result->setItemQty($qty);
```

Comment:
```suggestion
            $result->setItemQty((int)$qty ?: 1);
```

Reply:
@orlangur  `$qty` variable used below in file many times. I think it should contain already fixed value, isn't it?

Reply:
@likemusic it is casted to int only here within condition. Please don't forget to squash.

Reply:
@orlangur I think all other validation should by with casted value.
Assume $qty = 0.1 and $stockItem->getMinSaleQty() is 1. In $result we have 1.
Look at line 130 `if ($stockItem->getMinSaleQty() && $qty < $stockItem->getMinSaleQty()) {`

If here $qty == 1 will not add error message to $result. If $qty == 0.1 here, we will add error message to $result.
In latest case finaly we will have result with $qty = 1 but with massage that 'The fewest you may purchase is 1.'. It's not logical. Imho all validation with $qty variable in this method should be done with cased to int value.

Reply:
@likemusic ok, agree, assignment should still be there and `$result->setItemQty($qty = (int)$qty ?: 1);` would look awkward.

Reply:
I made:
```
git rebase -i 2.3-develop
git push -f
```

Is it right way to squash commits in pull-request?